src/link.c

```diff
@@ -247,6 +247,7 @@ bool8 IsWirelessAdapterConnected(void)
     SetWirelessCommType0_Internal();
     CloseLink();
     RestoreSerialTimer3IntrHandlers();
+    FlagSet(FLAG_NURSE_UNION_ROOM_REMINDER);
```

Comment:
Is this change necessary?

Reply:
Answering https://github.com/rh-hideout/pokeemerald-expansion/pull/5050#discussion_r1694220956 as well

When this is run for `OW_UNION_CONNECTED_ONLY_CHECK`, that flag is what prevents the nurse from trying to tell you about it. However, because Game Freak hates us, the game will still try to copy the name of a player from the Union Room. Because the Union Room was never initialized, it will try to do a StringCopy from an empty place, which will crash the game. Returning FALSE prevents that crash.